src/chainstate/stacks/db/blocks.rs

```diff
@@ -5070,6 +5071,7 @@ impl StacksChainState {
         chainstate_tx: &'b mut ChainstateTx,
         clarity_instance: &'a mut ClarityInstance,
         burn_dbconn: &'b dyn BurnStateDB,
+        sortition_dbconn: &'b dyn SortitionDBRef,
```

Comment:
Is there a world in which `burn_dbconn` and `sortition_dbconn` are different objects?  Why can't we implement `SortitionDBRef` for the same things we implement `BurnStateDB` for?

Reply:
Not really, but the reason for `SortitionDBRef` is that I didn't want to add these functions to  a `libclarity` interface (`SortitionDBRef` is defined in the main library, not `libclarity`). Ideally we'd be able to do something like trait upcasting, but it's not a thing in rust ([yet?](https://github.com/rust-lang/rust/issues/65991))

Reply:
Ah, gotcha.  Can you add a comment to the docstring that these are expected to be the same object (and maybe there's a way to test that somehow at runtime)?

Reply:
Will add a docstring. I don't think there's a way to do a reference equality check on the trait references without some unsafe code. This is because the pointers are in fact not the same: they are two different fat pointers, so to do the reference equality check, we'd have to (using `unsafe`) grab the bits of the pointer that comprise the content address and then compare those.